src/parsing/scope.ml

```diff
@@ -66,12 +66,9 @@ type lhs_data =
 
 type mode =
   | M_Term of
-      { mutable m_term_new_metas : problem
-      (** Metavariables generated during scoping. *)
-      ; m_term_meta_of_name : string -> meta option
-      (** Returns the meta with the given name. *)
-      ; m_term_meta_of_key : int -> meta option
-      (** Returns the meta with the give key. *)
+      { mutable m_term_placeholders : meta_ident list
+      (** Named placeholders. Used to ensure they do not appear non
+          linearly. *)
```

Comment:
Could you please recall why you need this linearity condition?

Reply:
The first possible answer is that it may be the case that non linear meta variables are useless. We can also say that it's dangerous to allow non linear meta variables because we don't know what happen if they are not in the same context, possibly the unification will just fail because it won't be able to instantiate the variable (it's just a guess).

Reply:
I think I can live with this restriction. It's just that it would be good to add a comment here.